docs/resources/credential_ssh_private_key.md

```diff
@@ -0,0 +1,74 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "boundary_credential_ssh_private_key Resource - terraform-provider-boundary"
+subcategory: ""
+description: |-
+  The SSH private key credential resource allows you to configure a credential using a username, private key and optional passphrase.
+---
+
+# boundary_credential_ssh_private_key (Resource)
+
+The SSH private key credential resource allows you to configure a credential using a username, private key and optional passphrase.
+
+## Example Usage
+
+```terraform
+resource "boundary_scope" "org" {
+  name                     = "organization_one"
+  description              = "global scope"
+  scope_id                 = "global"
+  auto_create_admin_role   = true
+  auto_create_default_role = true
+}
+
+resource "boundary_scope" "project" {
+  name                   = "project_one"
+  description            = "My first scope!"
+  scope_id               = boundary_scope.org.id
+  auto_create_admin_role = true
+}
+
+resource "boundary_credential_store_static" "example" {
+  name        = "example_static_credential_store"
+  description = "My first static credential store!"
+  scope_id    = boundary_scope.project.id
+}
+
+resource "boundary_credential_ssh_private_key" "example" {
+  name                   = "example_ssh_private_key"
+  description            =  "My first ssh private key credential!"
+  credential_store_id    = boundary_credential_store_static.example.id
+  username               = "my-username"
+  private_key            = "my-private-key"         # change to valid SSH Private Key
```

Comment:
I always struggle with how to input private key literals, so I think it would be great if we could have a real example here. Do I need to escape newlines? Can I load from a file? These are the sort of questions I'd _love_ an example to answer for me.

Reply:
Updated to show the how it would work with `file` which is the recommended way, to answer your question if you want to include it as a string I do believe you need to escape newlines.

Reply:
You can alternatively put in in an `EOT` so something like:
```
private_key                 = <<-EOT
            -----BEGIN OPENSSH PRIVATE KEY-----
            b3BlbnNzaC1rZXktdjEAAAAACmFlczI1Ni1jdHIAAAAGYmNyeXB0AAAAGAAAABCDScc/85
            T+BWUXGTm7e1P9AAAAEAAAAAEAAAAzAAAAC3NzaC1lZDI1NTE5AAAAINLEDJVqC3FRVsEh
            ZbkM+FkPvwGlhzdaRYzRQ2RU3ZuVAAAAsGhjh0SV2PY6F/su5OGvpI/ou/pzJY4VrESL/X
            g+7JBeOK4kWBo+SQnhzN+O8DK2DTpI+3q4xE4WwwR5E0tJlmCoLSpPYsVXi6eHRAQaiynQ
            cp1lXDlxEIsZKYmacOwyEF6kUUn4bHPU7zy7rU/z4QGqkPzWfl5ahe2hRx8a300oLAUCch
            pQ+Cdu6kUn4Cei/tQH1GwDxHevmRP1qwQ2H/u06T0GYlxc0IkDti9H62sP
            -----END OPENSSH PRIVATE KEY-----
        EOT
```